app/cdap/components/WrangleHome/Components/OngoingDataExplorations/index.tsx

```diff
@@ -0,0 +1,189 @@
+              workspaceId: eachValue.workspaceId,
+            };
+            const requestBody = {
+              directives: eachValue.directives,
+              limit: 1000,
+              insights: {
+                name: eachValue?.sampleSpec?.connectionName,
+                workspaceName: eachValue.workspaceName,
+                path: eachValue?.sampleSpec?.path,
+                visualization: {},
+              },
+            };
+
+            const connectorName: string | undefined = findConnectorType(
+              eachValue?.sampleSpec?.connectionName
+            );
+            if (connectorName) {
+              explorationData.push({
+                connectorType: connectorName,
+                connectionName: eachValue?.sampleSpec?.connectionName
+                  ? eachValue?.sampleSpec?.connectionName
+                  : T.translate(
+                      'features.WranglerNewUI.OnGoingDataExplorations.labels.importedDataset'
+                    ).toString(),
+                workspaceName: eachValue.workspaceName,
+                recipeSteps: eachValue.directives?.length ?? 0,
+                dataQuality: null,
+                workspaceId: eachValue.workspaceId,
+                count: 0,
+              });
+              return MyDataPrepApi.execute(params, requestBody);
+            }
+            return of(undefined);
+          });
+          return forkJoin(workspaces).pipe(defaultIfEmpty(null));
+        })
+      )
+      .subscribe((responses) => {
+        if (responses && Array.isArray(responses) && responses.length) {
+          responses
+            ?.filter((eachResponse) => eachResponse)
+            .forEach((workspace, index) => {
+              let dataQuality = 0;
+              workspace?.headers?.forEach((eachWorkspaceHeader) => {
+                const general = workspace.summary?.statistics[eachWorkspaceHeader]?.general;
+                // Here we are getting empty & non-null(renaming it as nonEmpty) values from general(API Response) and provinding default values for them
+                const { empty: empty = 0, 'non-null': nonEmpty = 100 } = general;
+
+                // Round number to next lowest .1%
+                // Number.toFixed() can round up and leave .0 on integers
+                const nonNull = Math.floor((nonEmpty - empty) * 10) / 10;
+                dataQuality = dataQuality + nonNull;
+              });
+              const totalDataQuality = dataQuality / workspace.headers?.length ?? 1;
+              explorationData[index].dataQuality = totalDataQuality;
+              explorationData[index].count = workspace.count;
+              const final = getUpdatedExplorationCards(explorationData);
+              setOnGoingExplorationsData(final);
+            });
+        }
+      });
+  }, []);
+
+  useEffect(() => {
+    getOngoingData();
+  }, []);
+
+  const filteredData = onGoingExplorationsData.filter(
+    (eachWorkspace) => eachWorkspace[6].count !== 0
```

Comment:
Why `6`?

Reply:
After data massaging, we are getting onGoingExplorationsData in an array of objects. And in that array of objects, the 7th object contains the count of columns present in a workspace. So, here we are just returning the workspaces which have data instead of returning all the workspaces(including workspaces which don't have any data).